mitmproxy.py

```diff
@@ -19,33 +21,56 @@ def binary(self):
             name = "mitmdump-osx"
         elif "win" in sys.platform:
             name = "mitmdump-win.exe"
-        return os.path.join(os.getcwd(), "utils", name)
+
+        return os.path.join(os.getcwd(), "utils", "mitm-{}".format(self.mitmproxy_version), name)
 
     def __enter__(self):
-        if self.record:
-            command = [
-                self.binary,
-                "--wfile",
-                self.path,
-                "--script",
-                " ".join(
-                    [
-                        os.path.join(self.scripts, "inject-deterministic.py")]
-                )
-            ]
+
+        if self.mitmproxy_version == "2.0.2":
+            if self.record:
+                command = [
+                    self.binary,
+                    "--wfile",
+                    self.path,
+                    "--script",
+                    " ".join(
+                        [
+                            os.path.join(self.scripts, "inject-deterministic.py")]
+                    )
+                ]
+            else:
+                command = [
+                    self.binary,
+                    "--replay-kill-extra",
+                    "--script",
+                    " ".join(
+                        [
+                            os.path.join(self.scripts, "alternate-server-replay.py"),
+                            self.path,
+                        ]
+                    ),
+                ]
+        elif self.mitmproxy_version == "4.0.4":
+            if self.record:
+                command = [
+                    self.binary,
+                    # "--scripts",
+                    # os.path.join(self.scripts, "inject-deterministic.py"),
```

Comment:
Why no deterministic in 4.0.4? I presume you need version specific versions of the script?

Reply:
We will need to update the addon to work with 4.0.4 
If we have the base ready it should not be a lot of work